targets/TARGET_GigaDevice/TARGET_GD32F30X/TATGET_GD32F307VG/PinNames.h

```diff
@@ -0,0 +1,239 @@
+    PORTB_9  = 0x19,
+    PORTB_9_MUL0 = PORTB_9 | MUL0,
+    PORTB_10 = 0x1A,
+    PORTB_11 = 0x1B,
+    PORTB_12 = 0x1C,
+    PORTB_13 = 0x1D,
+    PORTB_14 = 0x1E,
+    PORTB_15 = 0x1F,
+
+    PORTC_0  = 0x20,
+    PORTC_0_MUL0 = PORTC_0 | MUL0,
+    PORTC_1  = 0x21,
+    PORTC_1_MUL0 = PORTC_1 | MUL0,
+    PORTC_2  = 0x22,
+    PORTC_2_MUL0 = PORTC_2 | MUL0,
+    PORTC_3  = 0x23,
+    PORTC_3_MUL0 = PORTC_3 | MUL0,
+    PORTC_4  = 0x24,
+    PORTC_4_MUL0 = PORTC_4 | MUL0,
+    PORTC_5  = 0x25,
+    PORTC_5_MUL0 = PORTC_5 | MUL0,
+    PORTC_6  = 0x26,
+    PORTC_6_MUL0 = PORTC_6 | MUL0,
+    PORTC_7  = 0x27,
+    PORTC_7_MUL0 = PORTC_7 | MUL0,
+    PORTC_8  = 0x28,
+    PORTC_8_MUL0 = PORTC_8 | MUL0,
+    PORTC_9  = 0x29,
+    PORTC_9_MUL0 = PORTC_9 | MUL0,
+    PORTC_10 = 0x2A,
+    PORTC_11 = 0x2B,
+    PORTC_12 = 0x2C,
+    PORTC_13 = 0x2D,
+    PORTC_14 = 0x2E,
+    PORTC_15 = 0x2F,
+
+    PORTD_0  = 0x30,
+    PORTD_1  = 0x31,
+    PORTD_2  = 0x32,
+    PORTD_3  = 0x33,
+    PORTD_4  = 0x34,
+    PORTD_5  = 0x35,
+    PORTD_6  = 0x36,
+    PORTD_7  = 0x37,
+    PORTD_8  = 0x38,
+    PORTD_9  = 0x39,
+    PORTD_10 = 0x3A,
+    PORTD_11 = 0x3B,
+    PORTD_12 = 0x3C,
+    PORTD_13 = 0x3D,
+    PORTD_14 = 0x3E,
+    PORTD_15 = 0x3F,
+
+    PORTE_0  = 0x40,
+    PORTE_1  = 0x41,
+    PORTE_2  = 0x42,
+    PORTE_3  = 0x43,
+    PORTE_4  = 0x44,
+    PORTE_5  = 0x45,
+    PORTE_6  = 0x46,
+    PORTE_7  = 0x47,
+    PORTE_8  = 0x48,
+    PORTE_9  = 0x49,
+    PORTE_10 = 0x4A,
+    PORTE_11 = 0x4B,
+    PORTE_12 = 0x4C,
+    PORTE_13 = 0x4D,
+    PORTE_14 = 0x4E,
+    PORTE_15 = 0x4F,
+
+    /* ADC internal channels */
+    ADC_TEMP = 0xF0,
+    ADC_VREF = 0xF1,
+
+
+    /* Arduino connector namings */
+    A0          = PORTC_0,
+    A1          = PORTC_1,
+    A2          = PORTC_2,
+    A3          = PORTC_3,
+    A4          = PORTC_4,
+    A4_I2C_SDA  = PORTC_11,
+    A4_I2C_SCL  = PORTC_10,
+    A5          = PORTC_5,
+    D0          = PORTA_3,
+    D1          = PORTA_2,
+    D2          = PORTA_4,
+    D3          = PORTC_6,
+    D4          = PORTB_3,
+    D5          = PORTC_7,
+    D6          = PORTC_8,
+    D7          = PORTB_4,
+    D8          = PORTB_5,
+    D9          = PORTC_9,
+    D10         = PORTA_1,
+    D11         = PORTB_15,
+    D12         = PORTB_14,
+    D13         = PORTB_13,
+    D14         = PORTB_9,
+    D15         = PORTB_8,
+
+    LED1        = PORTE_0,
+    LED2        = PORTE_1,
+    LED3        = PORTE_6,
+
+    KEY2        = PORTA_0,
+    KEY3        = PORTB_1,
+    KEY4        = PORTD_2,
+
+    BUTTON1     = KEY2,
+    BUTTON2     = KEY3,
+    BUTTON3     = KEY4,
+
+    SERIAL_TX   = PORTA_9,
+    SERIAL_RX   = PORTA_10,
+    USBTX       = PORTA_9,
+    USBRX       = PORTA_10,
+
+    I2C_SCL     = PORTB_6,
+    I2C_SDA     = PORTB_7,
+    SPI_MOSI    = PORTA_7,
+    SPI_MISO    = PORTA_6,
+    SPI_SCK     = PORTA_5,
+    SPI_CS      = PORTE_3,
+    PWM_OUT     = PORTA_7,

```

Comment:
Only 1 of each peripheral brought out on the board?

Reply:
Yes, there are one SPI-Flash, one EEPROM on the board. The MCU has more than one peripheral, and meanwhile the pins are all out on the board.